src/tests/Microsoft.PowerFx.Core.Tests/BlobTests.cs

```diff
@@ -0,0 +1,138 @@
+
+            Assert.NotNull(blob);
+            Assert.Equal("Hello World!", blob.ResourceElement.GetAsStringAsync(null, CancellationToken.None).Result);
+            Assert.Equal("SGVsbG8gV29ybGQh", blob.ResourceElement.GetAsBase64Async(CancellationToken.None).Result);
+            Assert.Equal(new byte[] { 72, 101, 108, 108, 111, 32, 87, 111, 114, 108, 100, 33 }, blob.GetAsByteArrayAsync(CancellationToken.None).Result);
+            Assert.Same(blob.ResourceElement, sb);
+        }
+
+        [Fact]
+        public void BlobTest_SomeBase64Value()
+        {
+            BlobElementBase b64b = new Base64Blob("SGVsbG8gV29ybGQh");
+            BlobValue blob = new BlobValue(b64b);
+
+            Assert.NotNull(blob);
+            Assert.Equal("Hello World!", blob.ResourceElement.GetAsStringAsync(null, CancellationToken.None).Result);
+            Assert.Equal("SGVsbG8gV29ybGQh", blob.ResourceElement.GetAsBase64Async(CancellationToken.None).Result);
+            Assert.Equal(new byte[] { 72, 101, 108, 108, 111, 32, 87, 111, 114, 108, 100, 33 }, blob.GetAsByteArrayAsync(CancellationToken.None).Result);
+            Assert.Same(blob.ResourceElement, b64b);
+        }
+
+        [Fact]
+        public void BlobTest_SomeByteArrayValue()
+        {
+            BlobElementBase bab = new ByteArrayBlob(new byte[] { 72, 101, 108, 108, 111, 32, 87, 111, 114, 108, 100, 33 });
+            BlobValue blob = new BlobValue(bab);
+
+            Assert.NotNull(blob);
+            Assert.Equal("Hello World!", blob.ResourceElement.GetAsStringAsync(null, CancellationToken.None).Result);
+            Assert.Equal("SGVsbG8gV29ybGQh", blob.ResourceElement.GetAsBase64Async(CancellationToken.None).Result);
+            Assert.Equal(new byte[] { 72, 101, 108, 108, 111, 32, 87, 111, 114, 108, 100, 33 }, blob.GetAsByteArrayAsync(CancellationToken.None).Result);
+            Assert.Same(blob.ResourceElement, bab);
+        }
+
+        [Fact]
+        public void BlobTest_InvalidBase64()
+        {
+            ArgumentException ex = Assert.Throws<ArgumentException>(() => new Base64Blob("This_is_not_a_base64_string!"));
+            Assert.Equal("Invalid base64 string (Parameter 'base64Str')", ex.Message);
+        }
+
+        [Fact]
+        public void BlobTest_MultipleObjects()
+        {
+            BlobValue[] blobs = Enumerable.Range(0, 10).Select(i =>
+            {
+                BlobElementBase sre = new StringBlob($"Blob {i}");
+                return new BlobValue(sre);
+            }).ToArray();
+
+            int i = 0;
+            foreach (BlobValue blob in blobs)
+            {
+                Assert.NotNull(blob);
+                Assert.Equal($"Blob {i++}", blob.GetAsStringAsync(null, CancellationToken.None).Result);                
+            }          
+        }
+
+        [Fact]
+        public void BlobTest_LoadStream()
+        {
+            using Stream file = File.Open("icon.png", FileMode.Open, FileAccess.Read);
```

Comment:
>"icon.png" [](http://example.com/codeflow?start=42&length=10)

Instead of having a binary file we can have something like `var myStream = new MemoryStream(new byte[] { 0, 1, 2, 3, 4, 5, 6, 7 });`, use that stream and validate that you can retrieve the data using one of the GetAsXXX functions